test/AutoDiff/derivative_attr_type_checking.swift

```diff
@@ -532,3 +532,33 @@ struct StoredProperty: Differentiable {
     (stored, { _ in .zero })
   }
 }
+
+// When the generic signature is not considered while calculating the actual pullback type, the
+// typechecker does not realize that `T.TangentVector == Float`, and therefore it complains that
```

Comment:
The newly added tests mention that "the typechecker complains" but there are no `expected-error` lines.

Do these test newly fixed issues? If so, clarifying that in the comments would be nice (e.g. by using past tense, "the typechecker complained" and "The user could work around this").

Reply:
I'll change it to past tense. The correct behavior is no diagnostics. Before https://github.com/apple/swift/pull/28762, there were diagnostics.

Reply:
Got it, thanks!